src/protocol/vm/tycho_decoder.rs

```diff
@@ -122,9 +122,9 @@ impl TryFromWithBlock<ComponentWithState> for VMPoolState<PreCachedDB> {
                     .protocol_system
                     .as_str()
             });
-        let adapter_file_path =
-            format!("src/protocol/vm/assets/{}", to_adapter_file_name(protocol_name));
-        info!("Creating a new pool state for balancer pool with id {}", &id);
+        let mut adapter_file_path =
+            PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("src/protocol/vm/assets");
+        adapter_file_path = adapter_file_path.join(to_adapter_file_name(protocol_name));
```

Comment:
I think we can add the `.join(to_adapter_file_name(protocol_name))` above and not make it mutable